library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/recorder/SnapshotProducer.kt

```diff
@@ -6,13 +6,305 @@
+                )
+            }
+            TextView::class.java.isAssignableFrom(this::class.java) -> {
+                Node(
+                    id = resolvedId(),
+                    children = childNodes,
+                    wireframes = listOf((this as TextView).asTextWireframe(pixelsDensity))
+                )
+            }
+            ImageView::class.java.isAssignableFrom(this::class.java) -> {
+                Node(
+                    id = resolvedId(),
+                    children = childNodes,
+                    wireframes = listOf(
+                        (this as ImageView).asImageWireframeNode(pixelsDensity)
+                    )
+                )
+            }
+            else -> {
+                Node(
+                    id = resolvedId(),
+                    children = childNodes,
+                    wireframes = listOf(asShapeWireframe(pixelsDensity))
+                )
+            }
+        }
+
+    private fun View.isToolbar(): Boolean {
+        return (
+            Toolbar::class.java.isAssignableFrom(this::class.java) &&
+                this.id == androidx.appcompat.R.id.action_bar
+            ) ||
+            (
+                Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP &&
+                    android.widget.Toolbar::class.java.isAssignableFrom(
+                        this::class.java
+                    )
+                )
+    }
+
+    private fun View.isNotValid(): Boolean {
+        return !this.isShown || this.x < 0 || this.y < 0
+    }
+
+    private fun View.isSystemNoise(): Boolean {
+        return id in systemViewIds ||
+            this is ViewStub ||
+            this is ActionBarContextView
+    }
+
+    private fun View.resolvedId(): Long {
+        return if (id != View.NO_ID) id.toLong() else hashCode().toLong()
+    }
+
+    private fun View.asShapeWireframe(pixelsDensity: Float):
+        MobileSegment.Wireframe.ShapeWireframe {
+        val scaledHeight = height.densityNormalized(pixelsDensity).toLong()
+        val scaledWidth = width.densityNormalized(pixelsDensity).toLong()
+        val coordinates = IntArray(2)
+        // this will always have size >= 2
+        @Suppress("UnsafeThirdPartyFunctionCall")
+        this.getLocationInWindow(coordinates)
+        val x = coordinates[0].densityNormalized(pixelsDensity).toLong()
+        val y = coordinates[1].densityNormalized(pixelsDensity).toLong()
+        val styleBorderPair = background?.resolveShapeStyleAndBorder()
+        return MobileSegment.Wireframe.ShapeWireframe(
+            resolvedId(),
+            x,
+            y,
+            scaledWidth,
+            scaledHeight,
+            shapeStyle = styleBorderPair?.first,
+            border = styleBorderPair?.second
+        )
+    }
+
+    private fun Drawable.resolveShapeStyleAndBorder(): Pair<MobileSegment
+        .ShapeStyle?, MobileSegment.ShapeBorder?>? {
```

Comment:
Can we pick a consistent name for all those conversion functions? You have at least three in this class: `toSomething`, `asSomething` and `resolveSomething`.